src/Service/PdfCreator.php

```diff
@@ -4,38 +4,32 @@
 
 namespace App\Service;
 
-use Spipu\Html2Pdf\Html2Pdf;
+use App\ValueObject\PdfFormat;
+use App\ValueObject\PdfOrientation;
+use Dompdf\Dompdf;
+use Dompdf\Options;
 
 final class PdfCreator implements PdfCreatorInterface
 {
     private $pdf;
     private $orientation;
     private $format;
-    private $lang;
-    private $unicode;
-    private $encoding;
-    private $margin;
+    private $options;
 
-    public function __construct(string $orientation, string $format, string $lang, bool $unicode, string $encoding, array $margin)
+    public function __construct(PdfOrientation $orientation, PdfFormat $format, array $options)
     {
         $this->orientation = $orientation;
         $this->format = $format;
-        $this->lang = $lang;
-        $this->unicode = $unicode;
-        $this->encoding = $encoding;
-        $this->margin = $margin;
+        $this->options = $options;
     }
 
-    public function create(): void
+    public function generatePdf(string $template): string
```

Comment:
I was just wondering, is `$template` the whole html with replaced variables and all?

Reply:
Yes why ?

Reply:
Because I wasn't sure why we would not have the twig dependency and an array of parameters here.
So as it happens, the controller generates the template, which is IMO not his responsibility.
I'm not sure whether the service should know about twig and then the generatePdf changed to `$templatePath, $parameters`, or whether an extra service should be used (in hexagonal architecture that would be the command handler)... Maybe just leave it as is for now.